lightning/src/chain/keysinterface.rs

```diff
@@ -51,21 +51,29 @@ pub enum SpendableOutputDescriptor {
 		output: TxOut,
 	},
 	/// An output to a P2WSH script which can be spent with a single signature after a CSV delay.
-	/// The private key which should be used to sign the transaction is provided, as well as the
-	/// full witness redeemScript which is hashed in the output script_pubkey.
+	///
 	/// The witness in the spending input should be:
-	/// <BIP 143 signature generated with the given key> <empty vector> (MINIMALIF standard rule)
-	/// <witness_script as provided>
-	/// Note that the nSequence field in the input must be set to_self_delay (which corresponds to
-	/// the transaction not being broadcastable until at least to_self_delay blocks after the input
-	/// confirms).
+	/// <BIP 143 signature> <empty vector> (MINIMALIF standard rule) <provided witnessScript>
+	///
+	/// Note that the nSequence field in the spending transaction input must be set
+	/// to_self_delay (which corresponds to the transaction not being broadcastable until at
+	/// least to_self_delay blocks after the input confirms).
+	///
 	/// These are generally the result of a "revocable" output to us, spendable only by us unless
 	/// it is an output from us having broadcast an old state (which should never happen).
+	///
+	/// WitnessScript may be regenerated by passing the revocation_pubkey, to_self_delay and
```

Comment:
Can you mention somewhere that these are the things that are given by the users' ChannelKeys if they're using a custom ChannelKeys? I dont understand how a user using KeysManager/InMemoryChannelKeys can get this info, though? AFAICT, using those its entirely impossible to spend funds from such an event.

Reply:
Right, added a commit commenting this far better, and from where you should get each piece of the witnessScript. Tell me if you think it's good enough.